### PR TITLE
feat: backport changes to develop

### PR DESCRIPTION
**Pull request by tamtamchik**
It's just a backport of changes from the `master` branch to `develop`.

>[!IMPORTANT]
> Please do the following after update:
> - update node modules: `yarn`
> - update .env file (see `.env.example`)

### CODE REVIEW

**Review by rnmsslido — APPROVED** [review]
LGTM

**Review by TheDZhon — APPROVED** [review]
LGTM :+1:

File: lib/log.ts [hunk 1]
```diff
@@ -14,14 +19,13 @@ export const mg = (s: ConvertibleToString) => chalk.magenta(s);
...
 
-const INDENT = "    ";
-
 const MIN_LINE_LENGTH = 4;
 const LINE_LENGTH = 20;
 const LONG_LINE_LENGTH = 40;
 
 export const OK = gr("[✓]");
 export const NOT_OK = rd("[×]");
+export const WARN = yl("[!]");
```
[the remark this answers is not in the document]
Comment (tamtamchik) [reply lib/log.ts:28]: Correct me if I'm wrong, but I don't think it is some existing notation. 
It looks more like something internal, and we are free to use anything.

Maybe?

export const OK = gr(" ✅ ");
export const NOT_OK = rd(" 🚨 ");
export const WARN = yl(" ⚠️ ");

[the remark this answers is not in the document]
Comment (tamtamchik) [reply lib/log.ts:28]: Found source: https://www.npmjs.com/package/log-symbols

[the remark this answers is not in the document]
Comment (TheDZhon) [reply lib/log.ts:28]: Even better :)

File: scripts/dao-upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+# Compile contracts
+yarn compile
+
+# Generic migration steps file
+export STEPS_FILE=upgrade/steps.json
+
+yarn hardhat --network $NETWORK run --no-compile scripts/utils/migrate.ts
+
+# TODO
```
[the remark this answers is not in the document]
Comment (tamtamchik) [reply scripts/dao-upgrade.sh]: Fixed in eaaed4bfe4a0796169c4dfe7e07a432358bf4c01

File: lib/log.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tamtamchik) [reply lib/log.ts]: Fixed in eaaed4bfe4a0796169c4dfe7e07a432358bf4c01

File: .github/workflows/tests-integration-mainnet.yml [hunk 3]
```diff
@@ -3,14 +3,14 @@ name: Integration Tests
...
-    name: Hardhat
+  test_hardhat_integration_fork:
+    name: Hardhat / Mainnet
     runs-on: ubuntu-latest
     timeout-minutes: 120
 
     services:
       hardhat-node:
-        image: feofanov/hardhat-node:2.22.8
+        image: feofanov/hardhat-node:2.22.9
```
[the remark this answers is not in the document]
Comment (tamtamchik) [reply .github/workflows/tests-integration-mainnet.yml:13]: Working on it )

**Pull request merged**